src/Microsoft.Kusto.ServiceLayer/DataSource/DataSourceFactory.cs

```diff
@@ -9,9 +10,10 @@
 
 namespace Microsoft.Kusto.ServiceLayer.DataSource
 {
-    public class DataSourceFactory
+    [Export(typeof(IDataSourceFactory))]
+    public class DataSourceFactory : IDataSourceFactory
     {
-        public static IDataSource Create(DataSourceType dataSourceType, string connectionString, string azureAccountToken)
+        public IDataSource Create(DataSourceType dataSourceType, string connectionString, string azureAccountToken)
```

Comment:
This class was refactored to be injected instead of static because the calls would attempt to connect to a Kusto cluster when trying to unit test.